src/utils/calculations.ts

```diff
@@ -0,0 +1,153 @@
+    if (currentValue?.plannedBudget) {
+      return accumulator + currentValue.plannedBudget;
+    }
+    return accumulator;
+  }, 0);
+
+  const sumOfFrameBudgets = Object.values(rest).reduce((accumulator, currentValue) => {
+    if (currentValue?.frameBudget) {
+      return accumulator + currentValue.frameBudget;
+    }
+    return accumulator;
+  }, 0);
+
+  const sumOfFramesAndOverrun = sumOfFrameBudgets + budgetOverrunAmount;
+  const deviationBetweenCostEstimateAndBudget = sumOfFramesAndOverrun - sumOfPlannedBudgets;
+
+  const plannedBudgets = formatNumber(sumOfPlannedBudgets);
+  const costEstimateBudget = formatNumber(
+    type === 'group' ? projectBudgets : sumOfFramesAndOverrun,
+  );
+
+  return {
+    ...(type !== 'division' && {
+      plannedBudgets,
+      costEstimateBudget,
+      ...(type !== 'group' && {
+        deviation: formatNumber(deviationBetweenCostEstimateAndBudget),
+      }),
+    }),
+  };
+};
+
+/**
+ * Calculates the budgets for each cell for the current row. Return only the key for division and
+ * no deviation for groups.
+ *
+ * @returns
+ * - key: the key for the cell (key of the finance object)
+ * - plannedBudget: sum of all underlying project finances for that year (calculated in the backend)
+ * - frameBudget: the amount of frameBudget given to the underlying projects during that year
+ * - deviation: the deviation between the plannedBudget and the framedBudget and if the value is negative
+ */
+export const calculatePlanningCells = (
+  finances: IClassFinances,
+  type: PlanningRowType,
+): Array<IPlanningCell> => {
+  const { year, budgetOverrunAmount, projectBudgets, ...rest } = finances;
+  return Object.entries(rest).map(([key, value]) => {
+    const { frameBudget, plannedBudget } = value;
+    const deviation = frameBudget - plannedBudget;
+
+    return {
+      key,
+      // we don't return any budgets for divisions
+      ...(type !== 'division' && {
+        plannedBudget: formatNumber(plannedBudget),
+        // we don't return frameBudget or deviation for a group
+        ...(type !== 'group' && {
+          frameBudget: formatNumber(frameBudget),
+          deviation: formatNumber(deviation),
+        }),
+      }),
+    };
+  });
+};
+
+/**
+ * Takes an array of classes and sums together all their finances and create a list of planning cells with it.
+ *
+ * @returns
+ * - key: the key for the cell (key of the finance object)
+ * - plannedBudget: sum of all underlying project finances for that year (calculated in the backend)
+ * - frameBudget: the amount of frameBudget given to the underlying projects during that year
+ * - deviation: the deviation between the plannedBudget and the framedBudget and if the value is negative
+ */
+export const calculatePlanningSummaryCells = (
+  classes: Array<IClass>,
+  type: PlanningRowType,
+): Array<IPlanningCell> => {
+  const totalFinances = classes.reduce((acc: IClassFinances, curr: IClass) => {
```

Comment:
Rename acc and curr arguments to appropriate names

Reply:
fixed